### PR TITLE
Add test to arm.winedbg

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Some problems noticed when applying @Lazula 's pj patches.
* The return value of `pj_drain` should be manually freed, so better use `pj_string` instead.
* be_arm.c produces something like `ldmdbeq r2, {r0, r1}`, which is not json so pj cannot be applied.
* arm.winedbg lacks tests. I added some from db/asm/arm_32 and find bugs on `msr`, `mrc/mcr` and `cdp`. I fixed them according to [ARM datasheet](https://developer.arm.com/documentation/ddi0406/latest/) B9.3.10, B9.3.11, A8.8.112, A8.8.108 and A8.8.30.
* Removed 3 duplicate tests in db/asm/arm_32

### CODE REVIEW

File: libr/asm/arch/arm/winedbg/be_arm.c [hunk 1]
```diff
@@ -466,26 +442,14 @@ static ut16 thumb_disasm_pushpop(struct winedbg_arm_insn *arminsn, ut16 inst) {
...
 
-	for (i=0;i<=7;i++) {
-		if ((inst>>i) & 1) {
-			if (i == last) {
-				arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s", tbl_regs[i]);
-			}
-			else {
-				arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s, ", tbl_regs[i]);
-			}
+	for (i = 0; i <= 7; i++, inst >>= 1) {
```
Comment (trufae) [inline libr/asm/arch/arm/winedbg/be_arm.c:449]: What about < 8

File: libr/asm/arch/arm/winedbg/be_arm.c [hunk 2]
```diff
@@ -433,30 +421,18 @@ static ut16 thumb_disasm_pushpop(struct winedbg_arm_insn *arminsn, ut16 inst) {
...
```
Comment (trufae) [inline libr/asm/arch/arm/winedbg/be_arm.c:428]: I<8

File: libr/asm/arch/arm/winedbg/be_arm.c [hunk 3]
```diff
@@ -466,26 +442,14 @@ static ut16 thumb_disasm_pushpop(struct winedbg_arm_insn *arminsn, ut16 inst) {
...
+		if (inst & 1) {
+			arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s%s", first ? "" : ", ", tbl_regs[i]);
```
Comment (trufae) [inline libr/asm/arch/arm/winedbg/be_arm.c:451]: Rstrget in this first var

**Review by trufae — APPROVED** [review]

File: libr/asm/arch/arm/winedbg/be_arm.c (hunk 3 above)
Comment (eagleoflqj) [reply libr/asm/arch/arm/winedbg/be_arm.c:451]: > Rstrget in this first var

https://github.com/radareorg/radare2/blob/44755fe08d53b7426bbd33c2b5a0e0796d5cc343/libr/util/str.c#L823-L825
Not applicable.

**Pull request merged**